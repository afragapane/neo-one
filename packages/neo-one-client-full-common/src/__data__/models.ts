import { common, ContractParameterTypeModel, ECPoint, SignatureString, UInt160 } from '@neo-one/client-common';
import { constants } from '@neo-one/utils';
import {
  ContractABIModel,
  ContractEventModel,
  ContractGroupModel,
  ContractManifestModel,
  ContractMethodDescriptorModel,
  ContractModel,
  ContractParameterDeclarationModel,
  ContractPermissionDescriptorModel,
  ContractPermissionsModel,
  ContractPropertyStateModel,
} from '../models';

export const contractParamModel = {
  boolean: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Boolean,
    name: 'param',
  }),
  byteArray: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.ByteArray,
    name: 'param',
  }),
  hash160: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Hash160,
    name: 'param',
  }),
  hash256: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Hash256,
    name: 'param',
  }),
  array: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Array,
    name: 'param',
  }),
  integer: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Integer,
    name: 'param',
  }),
  interopInterface: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.InteropInterface,
    name: 'param',
  }),
  map: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Map,
    name: 'param',
  }),
  publicKey: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.PublicKey,
    name: 'param',
  }),
  signature: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Signature,
    name: 'param',
  }),
  string: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.String,
    name: 'param',
  }),
  void: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Void,
    name: 'param',
  }),
};

export const contractMethodDescriptorModel = (
  parameters: readonly ContractParameterDeclarationModel[] = [contractParamModel.boolean],
  returnType: ContractParameterTypeModel = ContractParameterTypeModel.Void,
  name = 'function',
) => new ContractMethodDescriptorModel({ name, parameters, returnType });

export const contractEventModel = (
  parameters: readonly ContractParameterDeclarationModel[] = [contractParamModel.boolean],
  name = 'event',
) => new ContractEventModel({ name, parameters });

export const contractAbiModel = (
  methods: readonly ContractMethodDescriptorModel[] = [contractMethodDescriptorModel()],
  events: readonly ContractEventModel[] = [contractEventModel()],
  entryPoint: ContractMethodDescriptorModel = contractMethodDescriptorModel(),
  hash = common.bufferToUInt160(Buffer.alloc(20, 1)),
) => new ContractABIModel({ hash, entryPoint, methods, events });

export const contractGroupModel = (
  publicKey: ECPoint = common.stringToECPoint(constants.PRIVATE_NET_PUBLIC_KEY),
  signature: SignatureString = 'ccaab040cc25021c91567b75db4778853441869157b8f6aad960cdcf1069812480027a528ca9b98e2205027de20696f848cf81824eeb7af1d5110870870ceb67',
) => new ContractGroupModel({ publicKey, signature });

export const contractPermissionDescriptorModel = (hashOrGroupType: 'uint160' | 'ecpoint' | undefined) => {
  let hashOrGroup: UInt160 | ECPoint | undefined;
  if (hashOrGroupType === 'uint160') {
    hashOrGroup = common.bufferToUInt160(Buffer.alloc(20, 1));
  } else if (hashOrGroupType === 'ecpoint') {
    hashOrGroup = common.stringToECPoint(constants.PRIVATE_NET_PUBLIC_KEY);
  }

  return new ContractPermissionDescriptorModel({ hashOrGroup });
};

export const contractPermissionsModel = (
  hashOrGroupType: 'uint160' | 'ecpoint' | undefined,
  methods: readonly string[] = [],
) => new ContractPermissionsModel({ contract: contractPermissionDescriptorModel(hashOrGroupType), methods });

export const contractManifestModel = (
  groups: readonly ContractGroupModel[] = [contractGroupModel()],
  features: ContractPropertyStateModel = ContractPropertyStateModel.HasStoragePayable,
  abi: ContractABIModel = contractAbiModel(),
  permissions: readonly ContractPermissionsModel[] = [contractPermissionsModel('uint160', ['method1'])],
  trusts: readonly UInt160[] = [common.bufferToUInt160(Buffer.alloc(20, 1))],
  safeMethods: readonly string[] = ['method1', 'method2'],
) => new ContractManifestModel({ groups, features, abi, permissions, trusts, safeMethods });

export const contractModel = (
  script: Buffer = Buffer.alloc(25),
  manifest: ContractManifestModel = contractManifestModel(),
) => new ContractModel({ script, manifest });
