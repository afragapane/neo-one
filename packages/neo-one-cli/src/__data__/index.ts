export * from './getClients';
export * from './getContracts';
