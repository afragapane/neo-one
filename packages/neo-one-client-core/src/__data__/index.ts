export * from './keys';
export * from './data';
