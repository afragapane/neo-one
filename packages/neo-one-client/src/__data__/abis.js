/* @flow */
import type {
  ABIEvent,
  ABIFunction,
  ABIParameter,
  ABIReturn,
  ABI,
} from '../types';

export const returns = {
  Signature: { type: 'Signature' },
  Boolean: { type: 'Boolean' },
  Hash160: { type: 'Hash160' },
  Hash256: { type: 'Hash256' },
  ByteArray: { type: 'ByteArray' },
  PublicKey: { type: 'PublicKey' },
  String: { type: 'String' },
  InteropInterface: { type: 'InteropInterface' },
  Void: { type: 'Void' },
  Integer: { type: 'Integer', decimals: 7 },
  Array: { type: 'Array', value: { type: 'String' } },
};

export const parameters = {
  Signature: { name: 'signatureName', type: 'Signature' },
  Boolean: { name: 'booleanName', type: 'Boolean' },
  Integer: { name: 'intName', type: 'Integer', decimals: 7 },
  Hash160: { name: 'hash160Name', type: 'Hash160' },
  Hash256: { name: 'hash256Name', type: 'Hash256' },
  ByteArray: { name: 'byteName', type: 'ByteArray' },
  PublicKey: { name: 'pubkeyName', type: 'PublicKey' },
  String: { name: 'stringName', type: 'String' },
  Array: {
    name: 'arrayName',
    type: 'Array',
    value: {
      name: 'stringName',
      type: 'String',
    },
  },
  InteropInterface: { name: 'interopName', type: 'InteropInterface' },
  Void: { name: 'voidName', type: 'Void' },
};

export const abiFunction = (
  constant: boolean = false,
  params: Array<ABIParameter> = [parameters.String],
  verify: boolean = false,
  name: string = 'funcName',
  returnType: ABIReturn = returns.Void,
): ABIFunction => ({
  name,
  constant,
  parameters: params,
  verify,
  returnType,
});

export const abiEvent = (
  name: string = 'eventName',
  params: Array<ABIParameter> = [parameters.String],
): ABIEvent => ({
  name,
  parameters: params,
});

export const abi = (
  functions: Array<ABIFunction> = [abiFunction()],
  events?: Array<ABIEvent> = [],
): ABI => ({
  functions,
  events,
});
