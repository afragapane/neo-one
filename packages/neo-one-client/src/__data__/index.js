/* @flow */
export { default as keys } from './keys';
export { default as transactions } from './transactions';
