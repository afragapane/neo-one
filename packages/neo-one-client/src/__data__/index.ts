export * from './keys';
export * from './data';
export * from './transactions';
export * from './factory';
