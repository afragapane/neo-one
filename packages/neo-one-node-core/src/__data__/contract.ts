import { common, ECPoint, SignatureString, UInt160 } from '@neo-one/client-common';
import { constants } from '@neo-one/utils';
import {
  Contract,
  ContractABI,
  ContractEvent,
  ContractGroup,
  ContractManifest,
  ContractMethodDescriptor,
  ContractParameterDeclaration,
  ContractParameterType,
  ContractPermissionDescriptor,
  ContractPermissions,
  ContractPropertyState,
} from '../contract';

export const testContext = { messageMagic: 0 };

export const contractParamDeclaration = {
  boolean: new ContractParameterDeclaration({
    type: ContractParameterType.Boolean,
    name: 'param',
  }),
  byteArray: new ContractParameterDeclaration({
    type: ContractParameterType.ByteArray,
    name: 'param',
  }),
  hash160: new ContractParameterDeclaration({
    type: ContractParameterType.Hash160,
    name: 'param',
  }),
  hash256: new ContractParameterDeclaration({
    type: ContractParameterType.Hash256,
    name: 'param',
  }),
  array: new ContractParameterDeclaration({
    type: ContractParameterType.Array,
    name: 'param',
  }),
  integer: new ContractParameterDeclaration({
    type: ContractParameterType.Integer,
    name: 'param',
  }),
  interopInterface: new ContractParameterDeclaration({
    type: ContractParameterType.InteropInterface,
    name: 'param',
  }),
  map: new ContractParameterDeclaration({
    type: ContractParameterType.Map,
    name: 'param',
  }),
  publicKey: new ContractParameterDeclaration({
    type: ContractParameterType.PublicKey,
    name: 'param',
  }),
  signature: new ContractParameterDeclaration({
    type: ContractParameterType.Signature,
    name: 'param',
  }),
  string: new ContractParameterDeclaration({
    type: ContractParameterType.String,
    name: 'param',
  }),
  void: new ContractParameterDeclaration({
    type: ContractParameterType.Void,
    name: 'param',
  }),
};

export const contractEvent = (
  parameters: readonly ContractParameterDeclaration[] = [contractParamDeclaration.boolean],
  name = 'event',
) =>
  new ContractEvent({
    name,
    parameters,
  });

export const contractMethodDescriptor = (
  parameters: readonly ContractParameterDeclaration[] = [contractParamDeclaration.boolean],
  returnType: ContractParameterType = ContractParameterType.Void,
  name = 'event',
) =>
  new ContractMethodDescriptor({
    name,
    parameters,
    returnType,
  });

export const contractAbi = (
  methods: readonly ContractMethodDescriptor[] = [contractMethodDescriptor()],
  events: readonly ContractEvent[] = [contractEvent()],
  entryPoint: ContractMethodDescriptor = contractMethodDescriptor(),
  hash = common.bufferToUInt160(Buffer.alloc(20, 1)),
) => new ContractABI({ hash, entryPoint, methods, events });

export const contractGroup = (
  publicKey: ECPoint = common.stringToECPoint(constants.PRIVATE_NET_PUBLIC_KEY),
  signature: SignatureString = 'ccaab040cc25021c91567b75db4778853441869157b8f6aad960cdcf1069812480027a528ca9b98e2205027de20696f848cf81824eeb7af1d5110870870ceb67',
) => new ContractGroup({ publicKey, signature });

export const contractPermissionDescriptor = (hashOrGroupType: 'uint160' | 'ecpoint' | undefined) => {
  let hashOrGroup: UInt160 | ECPoint | undefined;
  if (hashOrGroupType === 'uint160') {
    hashOrGroup = common.bufferToUInt160(Buffer.alloc(20, 1));
  } else if (hashOrGroupType === 'ecpoint') {
    hashOrGroup = common.stringToECPoint(constants.PRIVATE_NET_PUBLIC_KEY);
  }

  return new ContractPermissionDescriptor({ hashOrGroup });
};

export const contractPermissions = (
  hashOrGroupType: 'uint160' | 'ecpoint' | undefined,
  methods: readonly string[] = [],
) => new ContractPermissions({ contract: contractPermissionDescriptor(hashOrGroupType), methods });

export const contractManifest = (
  groups: readonly ContractGroup[] = [contractGroup()],
  features: ContractPropertyState = ContractPropertyState.HasStoragePayable,
  abi: ContractABI = contractAbi(),
  permissions: readonly ContractPermissions[] = [contractPermissions('uint160', ['method1'])],
  trusts: readonly UInt160[] = [common.bufferToUInt160(Buffer.alloc(20, 1))],
  safeMethods: readonly string[] = ['method1', 'method2'],
) => new ContractManifest({ groups, features, abi, permissions, trusts, safeMethods });

export const contract = (script: Buffer = Buffer.alloc(25), manifest: ContractManifest = contractManifest()) =>
  new Contract({ script, manifest });
