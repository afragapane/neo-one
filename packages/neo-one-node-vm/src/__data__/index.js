/* @flow */
export { default as createBlockchain } from './createBlockchain';
export { default as keys } from './keys';
export * as transactions from './transactions';
export * as testUtils from './utils';
