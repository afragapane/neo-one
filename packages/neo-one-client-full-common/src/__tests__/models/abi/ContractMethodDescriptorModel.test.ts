import { BinaryWriter, ContractParameterTypeModel } from '@neo-one/client-common';
import { contractMethodDescriptorModel, contractParamModel } from '../../../__data__';

describe('ContractMethodDescriptorModel - serializeWireBase', () => {
  let writer: BinaryWriter;
  beforeEach(() => {
    writer = new BinaryWriter();
  });

  test('Boolean', () => {
    contractMethodDescriptorModel().serializeWireBase(writer);
    expect(writer.toBuffer()).toMatchSnapshot();
  });

  test('Multiple Params', () => {
    contractMethodDescriptorModel(
      [contractParamModel.boolean, contractParamModel.integer, contractParamModel.string],
      ContractParameterTypeModel.String,
    ).serializeWireBase(writer);
    expect(writer.toBuffer()).toMatchSnapshot();
  });
});
