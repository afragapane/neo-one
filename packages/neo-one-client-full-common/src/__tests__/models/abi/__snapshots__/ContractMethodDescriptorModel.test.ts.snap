// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractMethodDescriptorModel - serializeWireBase Boolean 1`] = `"0866756e6374696f6e010105706172616dff"`;

exports[`ContractMethodDescriptorModel - serializeWireBase Multiple Params 1`] = `"0866756e6374696f6e030105706172616d0205706172616d0705706172616d07"`;
